Lib/test/test_imaplib.py

```diff
@@ -458,18 +458,15 @@ def test_simple_with_statement(self):
         with self.imap_class(*server.server_address):
             pass
 
-    @requires_resource('walltime')
     def test_imaplib_timeout_test(self):
-        _, server = self._setup(SimpleIMAPHandler)
-        addr = server.server_address[1]
-        client = self.imap_class("localhost", addr, timeout=None)
-        self.assertEqual(client.sock.timeout, None)
-        client.shutdown()
-        client = self.imap_class("localhost", addr, timeout=support.LOOPBACK_TIMEOUT)
-        self.assertEqual(client.sock.timeout, support.LOOPBACK_TIMEOUT)
-        client.shutdown()
+        _, server = self._setup(SimpleIMAPHandler, connect=False)
+        with self.imap_class(*server.server_address, timeout=None) as client:
+            self.assertEqual(client.sock.timeout, None)
+        with self.imap_class(*server.server_address, timeout=support.LOOPBACK_TIMEOUT) as client:
+            self.assertEqual(client.sock.timeout, support.LOOPBACK_TIMEOUT)
         with self.assertRaises(ValueError):
-            client = self.imap_class("localhost", addr, timeout=0)
+            with self.imap_class(*server.server_address, timeout=0):
+                pass
```

Comment:
It is not clear where the exception was raised: in `imap_class()`, in `__enter__()` or in `__exit__()`? I suggest to keep a simple `imap_class()` call here. No need to assign the result to a variable, as this never happens.
```suggestion
            self.imap_class(*server.server_address, timeout=0)
```